com.avaloq.tools.ddk.xtext.builder/src/com/avaloq/tools/ddk/xtext/builder/MonitoredClusteringBuilderState.java

```diff
@@ -928,7 +942,10 @@ protected boolean recordDeltaAsNew(final Delta newDelta) {
         if (!loadingStrategy.mayProcessAnotherResource(resourceSet, resourceSet.getResources().size())) {

```

Comment:
|| resourceToRetry == null 

I would also check if we're retrying. Asking runtime for the information is not so exact. So if we failed because of memory, we need to clear it.